src/crypto-api.ts

```diff
@@ -713,7 +713,8 @@ export interface CrossSigningKeyInfo {
  * Recovery key created by {@link CryptoApi#createRecoveryKeyFromPassphrase}
  */
 export interface GeneratedSecretStorageKey {
-    keyInfo?: AddSecretStorageKeyOpts;
+    /** Information to generate the key from a passphrase if any. */
+    passphrase?: PassphraseInfo;
```

Comment:
The problem with this change (replacing the optional `keyInfo` property with an optional `passphrase` property)   is that it means any existing implementation of `CreateSecretStorageOpts.createSecretStorageKey` (such as the one in matrix-react-sdk) will be incorrect. Worse, it will fail *silently* because new new property is optional. I think this is very dangerous.

I see two possible ways forward:

 * Retain the `keyInfo` property, but with a reduced structure. This is what I suggested before:
    ```ts
    export interface GeneratedSecretStorageKey {
        keyInfo?: {
            passphrase?: PassphraseInfo;
            name?: string;  // maybe we don't need this?
        },
        privateKey: Uint8Array;
        encodedPrivateKey?: string;
    }
    ```
 * Define a new interface to replace `GeneratedSecretStorageKey` (and deprecate the old one). Define a new property in `CreateSecretStorageOpts` to replace `createSecretStorageKey` which returns the new interface (and deprecate the old one); allow clients to provide either interface.

The second option ends up being more elegant (once we get rid of the deprecated classes/properties), but seems a lot more work.